core/store/src/db/testdb.rs

```diff
@@ -18,6 +25,9 @@ pub struct TestDB {
     // The TestDB doesn't produce any stats on its own, it's up to the user of
     // this class to set the stats as they need it.
     stats: RwLock<Option<StoreStatistics>>,
+
+    // Flags to change the default behavior, for testing purposes.
+    flags: RwLock<Option<TestStoreFlags>>,
```

Comment:
Why do you need RwLock and Option? 

RwLock -> Can you set it when the TestDB is created and leave it immutable? 
Option -> Can you add a sensible default and always provide it?

Reply:
I just followed how `stats` are done. But yeah, we do not need it mutable and can provide sensible default. Changing it.

Reply:
I followed how `stats` are done.
Also, did not want to change the `new()` signature as it is used in 144 places and this is likely just a temporary solution.
But yeah, flags could be immutable and we can provide sensible defaults, will add another version of `new()` and remove `set_store_flags`.